build/patches/experimental-user-scripts-support.patch

```diff
@@ -0,0 +1,10299 @@
++      render_frame);
++  }
+ }
+ 
+ void ChromeContentRendererClient::RunScriptsAtDocumentIdle(
+@@ -1488,6 +1515,12 @@ void ChromeContentRendererClient::RunScriptsAtDocumentIdle(
+       render_frame);
+   // |render_frame| might be dead by now.
+ #endif
++  user_scripts::UserScriptsRendererClient* userscript_client =
++    user_scripts::UserScriptsRendererClient::GetInstance();
++  if(userscript_client) {
++    userscript_client->RunScriptsAtDocumentIdle(
++      render_frame);
++  }
+ }
+ 
+ void ChromeContentRendererClient::
+diff --git a/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java b/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
+--- a/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
++++ b/components/browser_ui/settings/android/java/src/org/chromium/components/browser_ui/settings/SettingsUtils.java
+@@ -24,6 +24,12 @@ import androidx.preference.PreferenceFragmentCompat;
+ 
+ import org.chromium.base.ApiCompatibilityUtils;
+ 
++import org.chromium.base.Log;
++import androidx.preference.PreferenceScreen;
++import androidx.preference.PreferenceManager;
++import android.content.res.Resources;
++import java.lang.reflect.Field;
++
+ /**
+  * A helper class for Settings.
+  */
+@@ -126,4 +132,39 @@ public class SettingsUtils {
+         ImageView imageButton = (ImageView) button;
+         return imageButton.getDrawable() == parentMenu.getOverflowIcon();
+     }
++
++    public static PreferenceScreen inflatePrefsAddon(PreferenceFragmentCompat preferenceFragment,
++                                                     String name,
++                                                     Field[] fields) {
++        Resources resources = preferenceFragment.getResources();
++
++        for(int count=0; count < fields.length; count++) {
++            String assetName = fields[count].getName();
++            if (assetName.startsWith(name)) {
++                try {
++                    int resId = fields[count].getInt(null);
++                    return SettingsUtils.inflatePreferencesFromResource(preferenceFragment, resId);
++                } catch (IllegalAccessException e) {}
++            }
++        }
++        return null;
++    }
++
++    public static PreferenceScreen inflatePreferencesFromResource(
++            PreferenceFragmentCompat preferenceFragment, @XmlRes int preferencesResId) {
++        StrictMode.ThreadPolicy oldPolicy = StrictMode.allowThreadDiskReads();
++        try {
++            PreferenceManager preferenceManager = preferenceFragment.getPreferenceManager();
++            final PreferenceScreen xmlRoot = preferenceManager.inflateFromResource(
++                preferenceFragment.getContext(),
++                preferencesResId, preferenceFragment.getPreferenceScreen());
++            return xmlRoot;
++        } finally {
```

Comment:
Catch and log?

Reply:
no, the exception should just go up the stack